src/System.Private.Xml/src/System/Xml/Core/XmlWriter.cs

```diff
@@ -659,7 +659,25 @@ private void WriteLocalNamespaces(XPathNavigator nsNav)
         // Creates an XmlWriter for writing into the provided file.
         public static XmlWriter Create(string outputFileName)
         {
-            return Create(outputFileName, null);
+            if (outputFileName == null)
+            {
+                throw new ArgumentNullException(nameof(outputFileName));
+            }
+
+            // Avoid using XmlWriter.Create(string, XmlReaderSettings), as it references a lot of types
+            // that then can't be trimmed away.
+            var fs = new FileStream(outputFileName, FileMode.Create, FileAccess.Write, FileShare.Read);
+            try
+            {
+                var settings = new XmlWriterSettings() { CloseOutput = true };
+                XmlWriter writer = new XmlEncodedRawTextWriter(fs, settings);
+                return new XmlWellFormedWriter(writer, settings);
+            }
+            catch
```

Comment:
could this be just `finally`?

Reply:
No, we don't want to dispose of the file stream in the success case where it's wrapped by the writer we're successfully returning.